piduino/core/Arduino.c

```diff
@@ -390,6 +390,7 @@ static int map_registers(uint32_t reg_offset){
 void init_pins(){
   uint32_t revision = getBoardRev();
   if(revision == 0xa01041) revision = 0x10;//pi2 has B+ pinout
+  if(revision == 0xa21041) revision = 0x13;//pi2 has B+ pinout
```

Comment:
did this work for you? interesting why I had different number before? will check my PI2 and see if it's the same.